src/generic/Matrix.jl

```diff
@@ -10,16 +10,16 @@
 #
 ###############################################################################
 
-parent_type(::Type{S}) where {T <: NCRingElement, S <: Mat{T}} = MatSpace{T}
+parent_type(::Type{<:MatElem{T}}) where {T <: NCRingElement} = MatSpace{T, parent_type(T)}
 
-elem_type(::Type{MatSpace{T}}) where {T <: NCRingElement} = MatSpaceElem{T}
+elem_type(::Type{<:MatSpace{T}}) where {T <: NCRingElement} = dense_matrix_type(T)
```

Comment:
If I am not mistaken, this would change `elem_type(Generic.MatSpace{QQFieldElem})` from `Generic.MatSpaceElem{QQFieldElem}` to `QQMatrix`. Is this intended?

Reply:
Yes. At the moment every matrix type implement its own matrix space type, like `QQMatrix` comes with `QQMatrixSpace`. All those "space" types do the same. They just store `nrows`, `ncols` and the `base_ring`. So the idea is to replace all those individual types by a truly generic "space" type.